python/ray/train/batch_predictor.py

```diff
@@ -125,14 +156,19 @@ def __call__(self, batch):
         ray_remote_args["num_cpus"] = num_cpus_per_worker
         ray_remote_args["num_gpus"] = num_gpus_per_worker
 
-        return data.map_batches(
+        prediction_results = dropped_dataset.map_batches(
             ScoringWrapper,
             compute=compute,
             batch_format="pandas",
             batch_size=batch_size,
             **ray_remote_args,
         )
 
+        if original_col_ds:
+            prediction_results = prediction_results.zip(original_col_ds)
```

Comment:
im curious what guarantees ray datasets provide here, how did we ensure results returned from `dropped_dataset` and `original_col_ds` always match 1-1 across multiple executions, rather than assigning wrong? labels

Reply:
Hmm not sure I understand you here.

They don't necessarily need to match?